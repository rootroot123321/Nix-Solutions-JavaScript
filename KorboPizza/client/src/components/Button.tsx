import * as React from 'react';
import classNames from 'classnames';

interface ButtonProps {
    onClick?: () => void;
    type?: "button" | "submit" | "reset" | undefined;
    className?: string;
    outline?: boolean;
    children?: React.ReactNode;
}

const Button = ({onClick, type, className, outline, children}: ButtonProps) => {
    return (
        <button
            onClick={onClick}
            type={type ?? undefined}
            className={classNames('button', className, {
                'button--outline': outline,
            })}>
            {children}
        </button>
    );
};

export default Button;
