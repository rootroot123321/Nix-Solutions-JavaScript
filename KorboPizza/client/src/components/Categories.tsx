import * as React from 'react';

interface CategoriesProps {
    activeCategory: (null|number);
    items: Array<string>;
    onClickCategory: (index: (null|number)) => void;
}

const Categories: React.FC<CategoriesProps> = React.memo(function Categories({activeCategory = null, items = [], onClickCategory}: CategoriesProps) {
    return (
        <div className="categories">
            <ul>
                <li
                    className={activeCategory === null ? 'active' : ''}
                    onClick={() => onClickCategory(null)}>
                    Все
                </li>
                {items &&
                items.map((name, index) => (
                    <li
                        className={activeCategory === index + 1 ? 'active' : ''}
                        onClick={() => onClickCategory(index + 1)}
                        key={`${name}_${index}`}>
                        {name}
                    </li>
                ))}
            </ul>
        </div>
    );
});

export default Categories;
