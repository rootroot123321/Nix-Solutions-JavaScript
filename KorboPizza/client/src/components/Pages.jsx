import React from 'react';
import {useDispatch, useSelector} from "react-redux";
import {setPage} from "../redux/actions/pizzas";

const Pages = () => {
    const dispatch = useDispatch();
    const {page:currentPage, totalCount, limit} = useSelector(({pizzas}) => pizzas);
    const pageCount = Math.ceil(totalCount / limit);
    const pages = [];
    for(let i = 0; i < pageCount; i++) {
        pages.push(i + 1);
    }
    return (
        <div className="pages">
            {pages.map(page =>
                <div
                    key={page}
                    className={`page ${page === currentPage ? 'current_page' : ''}`}
                    onClick={() => dispatch(setPage(page))}>{page}</div>
            )}
        </div>
    );
};

export default Pages;