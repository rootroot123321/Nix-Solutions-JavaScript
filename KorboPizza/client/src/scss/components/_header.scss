@import '../variables';

.header {
  border-bottom: 1px solid $gray-line;
  padding: 40px 0;

  .container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    @media (max-width: 768px) {
      flex-direction: column;
    }
  }

  @media (max-width: 768px) {
    &__cart {
      margin-top: 30px;
    }
  }

  &__logo {
    display: flex;

    img {
      margin-right: 15px;
    }

    h1 {
      color: #181818;
      font-size: 24px;
      letter-spacing: 0.01em;
      text-transform: uppercase;
      font-weight: 800;
    }

    p {
      color: #7b7b7b;
    }
  }
  &__navbar {
    display: flex;
    align-items: center;
    button {
      border-radius: 10px;
      margin: 0 5px
    }
    select {
      width: 50px;
      height: 40px;
      font-size: 1.5em;
      background-color: $orange;
      border: none;
      border-radius: 10px;
      color: white;
    }
  }
}
