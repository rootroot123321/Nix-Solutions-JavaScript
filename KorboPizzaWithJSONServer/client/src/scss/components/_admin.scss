@import '../variables';

.admin {
  display: flex;
  flex-direction: column;
  margin: 5%;

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 3% 0;
    @mixin input-style {
      background-color: rgba(255, 155, 0, 0.5);
      width: 35%;
      height: 30px;
      padding: 2%;
      font-size: 1em;
      margin-bottom: 1.5%;
      border-radius: 10px;
    }
    .checkbox {
      display: flex;
      flex-direction: row;
    }
    input:not([type='file']) {
      @include input-style;
    }

    select {
      @include input-style;
      padding: initial;
    }
  }
}
