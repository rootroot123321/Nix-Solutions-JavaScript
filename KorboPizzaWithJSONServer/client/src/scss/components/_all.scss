@import './header';
@import './button';
@import './categories';
@import './sort';
@import './pizza-block';
@import './auth';
@import './admin';
