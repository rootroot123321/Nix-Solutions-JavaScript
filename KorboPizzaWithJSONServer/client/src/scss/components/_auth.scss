@import '../variables';

.authorization {
  border: 1px solid #ddd;
  background-color: rgba(255, 150, 0, 0.2);
  text-align: center;
  height: 120%;
  padding: 10%;

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 5%;

    input {
      background-color: rgba(255, 155, 0, 0.5);
      width: 50%;
      height: 50px;
      padding: 2%;
      font-size: 1.5em;
      margin: 1.5%;
      border-radius: 10px;
    }
    span {
      margin-top: 10px;
      a {
        color: dodgerblue;
        margin-left: 10px;
      }
    }
  }
}
