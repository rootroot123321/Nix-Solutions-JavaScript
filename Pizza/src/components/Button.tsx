import * as React from 'react';
import classNames from 'classnames';

interface ButtonProps {
    onClick?: () => void;
    className?: string;
    outline?: boolean;
    children?: React.ReactNode;
}

const Button = ({onClick, className, outline, children}: ButtonProps) => {
    return (
        <button
            onClick={onClick}
            className={classNames('button', className, {
                'button--outline': outline,
            })}>
            {children}
        </button>
    );
};

export default Button;
