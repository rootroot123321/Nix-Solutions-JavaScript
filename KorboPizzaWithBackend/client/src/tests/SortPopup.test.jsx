import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils";
import pretty from "pretty";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";

import {SortPopup} from "../components";
import store from "../redux/store";

let container = null;
const sortPopup = (activeSortType, items, onClickSortType) => {
    return (
        <Provider store={store}>
            <BrowserRouter>
                <SortPopup
                    activeSortType={activeSortType}
                    items={items}
                    onClickSortType={onClickSortType}
                />
            </BrowserRouter>
        </Provider>
    );
};
const fakePopup = [
    {name: "цене", type: "price", order: "desc"},
    {name: "алфавиту", type: "name", order: "asc"},
    {name: "популярности", type: "popular", order: "desc"},
    {name: "дате добавления", type: "added", order: "desc"},
    {name: "дате обновления", type: "updated", order: "desc"},
    {name: "качеству", type: "quality", order: "desc"},
    {name: "продаваемости", type: "sell", order: "asc"},
];
const onClickSortType = jest.fn();

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("renders the sort popup data (unit)", () => {
    act(() => {
        render(sortPopup("price", fakePopup, onClickSortType), container);
    });
    expect(
        container.querySelector("[data-testid=active-label]").textContent
    ).toBe("цене");
    const activeLabel = container.querySelector("[data-testid=active-label]");
    act(() => {
        activeLabel.dispatchEvent(new MouseEvent("click", {bubbles: true}));
    });
    for (let i = 0; i < fakePopup.length; i++) {
        expect(
            container.querySelector(".sort__popup > ul").children[i].textContent
        ).toBe(fakePopup[i].name);
    }
    const label = container.querySelector(".sort__popup > ul").children[3];
    act(() => {
        label.dispatchEvent(new MouseEvent("click", {bubbles: true}));
    });
    expect(onClickSortType).toHaveBeenCalledTimes(1);
    const next = fakePopup.filter((item) => item.name === label.textContent)[0];
    act(() => {
        render(sortPopup(next.type, fakePopup, onClickSortType), container);
    });
    expect(
        container.querySelector("[data-testid=active-label]").textContent
    ).toBe("дате добавления");
});

it("renders the sort popup data (snapshot)", () => {
    act(() => {
        render(sortPopup("name", fakePopup, onClickSortType), container);
    });
    expect(
        pretty(container.querySelector("[data-testid=active-label]").textContent)
    ).toMatchInlineSnapshot(`"алфавиту"`);
    const activeLabel = container.querySelector("[data-testid=active-label]");
    act(() => {
        activeLabel.dispatchEvent(new MouseEvent("click", {bubbles: true}));
    });
    for (let i = 0; i < fakePopup.length; i++) {
        expect(
            pretty(
                container.querySelector(".sort__popup > ul").children[i].textContent
            )
        ).toMatchSnapshot();
    }
    const label = container.querySelector(".sort__popup > ul").children[6];
    act(() => {
        label.dispatchEvent(new MouseEvent("click", {bubbles: true}));
    });
    const next = fakePopup.filter((item) => item.name === label.textContent)[0];
    act(() => {
        render(sortPopup(next.type, fakePopup, onClickSortType), container);
    });
    expect(
        pretty(container.querySelector("[data-testid=active-label]").textContent)
    ).toMatchInlineSnapshot(`"продаваемости"`);
});
