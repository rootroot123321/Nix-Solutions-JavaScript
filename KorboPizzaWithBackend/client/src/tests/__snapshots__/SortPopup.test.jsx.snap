// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the sort popup data (snapshot) 2`] = `"цене"`;

exports[`renders the sort popup data (snapshot) 3`] = `"алфавиту"`;

exports[`renders the sort popup data (snapshot) 4`] = `"популярности"`;

exports[`renders the sort popup data (snapshot) 5`] = `"дате добавления"`;

exports[`renders the sort popup data (snapshot) 6`] = `"дате обновления"`;

exports[`renders the sort popup data (snapshot) 7`] = `"качеству"`;

exports[`renders the sort popup data (snapshot) 8`] = `"продаваемости"`;
